src/FsAutoComplete/LspServers/AdaptiveFSharpLspServer.fs

```diff
@@ -2846,7 +2845,7 @@ type AdaptiveFSharpLspServer(workspaceLoader: IWorkspaceLoader, lspClient: FShar
                     Arguments = None }
 
                 return { p with Command = Some cmd } |> Some |> success
-            else
+            elif typ = "reference" then
```

Comment:
I figured we should guard here instead of just assume we want references.  I could also refactoring this code to do more of a dispatch on `typ` vs if statements.

Reply:
yes, 100%! I'm ok with `if` now, but a dispatch would be perfect. right now we only have 2 commands so we haven't really dug into a proper dispatch mechanism.